model/src/util.py

```diff
@@ -44,48 +80,65 @@ def apply(self, error: float, dt: float = 1) -> float:
 
 class SpikeMovingAverage:
 
-    def __init__(self, tau_mean: float) -> None:
+    def __init__(self, tau_mean: float = 600) -> None:
         """
         tau_mean:
          * A time constant that determines the smoothing factor for the moving average
            of spikes.
          * A smaller tau_mean will make the average more sensitive to recent spikes.
          * A larger tau_mean will give a smoother average that is less responsive to
            individual spikes, reflecting a longer-term average rate.
+
+        Zenke's paper uses a tau_mean of 600s
         """
-        self.mean: float = 0
+        self.mean: Optional[torch.Tensor] = None
         self.tau_mean = tau_mean
 
-    def apply(self, spike: float, dt: float = 1) -> float:
+    def apply(self, spike: torch.Tensor, dt: float = 1) -> torch.Tensor:
+        if self.mean is None:
+            # Initialize mean based on the first spike received
+            self.mean = torch.zeros_like(spike)
+
         # Apply the exponential decay to the mean state and add the new spike value
         self.mean += (dt / self.tau_mean) * (spike - self.mean)
         return self.mean
 
-    def tracked_value(self) -> float:
+    def tracked_value(self) -> torch.Tensor:
+        if self.mean is None:
+            raise ValueError("No data has been received yet")
+
         return self.mean
 
 
 class VarianceMovingAverage:
 
-    def __init__(self, tau_var: float) -> None:
+    def __init__(self, tau_var: float = .02) -> None:
```

Comment:
@mstormbull I added the unit descriptions to the bottom of docstring, and for moving averages (spike and variance), I actually assigned the default params. I selected to make one unit == 1 second. If you would rather do 1ms please let me know.

Reply:
Perfect, 1sec is a sensible choice.